libraries/rtos/rtx/TARGET_CORTEX_A/rt_CMSIS.c

```diff
@@ -905,7 +905,7 @@ uint8_t osThreadGetState (osThreadId thread_id) {
   if (__exceptional_mode()) return osErrorISR;     // Not allowed in ISR
 
   ptcb = rt_tid2ptcb(thread_id);                // Get TCB pointer
-  if (ptcb == NULL) return osErrorParameter;
+  if (ptcb == NULL) return INACTIVE;
```

Comment:
Note - In the new upcoming cmsis 5 , they declare ``osThreadState osThreadGetState (osThreadId thread_id);`` . I could not locate this function in the current cmsis repository (the version we are using here), thus this was an extension we provided.